stacks_queues/queue.py

```diff
@@ -15,47 +15,73 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
+        
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException("Queue is full")
+            
+        if self.size == 0:
+            self.front, self.rear = (0, 0)
+
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
+        self.size += 1    
+
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException("Queue is empty")
+        first_element = self.store[self.front]
+        self.front = (self.front + 1) % self.buffer_size
+        self.size -= 1
+        return first_element
+
 
     def front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
-        
+        if not self.store:
+            return None
+        return self.store[0]
+
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.store.size
```

Comment:
`self.store` does not have a `size` attribute - `self.store` is an array; we would use `len()` to find the size of `self.store`.  Moreover, we initialize `self.store` to be the maximum size our queue can be, aka our buffer size(`INITIAL_QUEUE_SIZE`), so the size of `self.store` never changes. 

Our `Queue` class does, however, have an attribute `size` which you increment/decrement every time you enqueue/dequeue an item respectively. Consider, using this `size` attribute to implement this method instead.